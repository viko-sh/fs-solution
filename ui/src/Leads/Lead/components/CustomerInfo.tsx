import React from 'react';
import { LeadProps } from '../type';
import { CustomerInfoWrapper } from '../style';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPhoneAlt, faEnvelope } from '@fortawesome/free-solid-svg-icons';

export const CustomerInfo = ({
  contactPhone,
  contactEmail
}: Partial<LeadProps>) => {
  return (
    <CustomerInfoWrapper>
      <ul>
        <li>
          <FontAwesomeIcon icon={faPhoneAlt} />
          {contactPhone}
        </li>
        <li>
          <FontAwesomeIcon icon={faEnvelope} />
          {contactEmail}
        </li>
      </ul>
    </CustomerInfoWrapper>
  );
};
