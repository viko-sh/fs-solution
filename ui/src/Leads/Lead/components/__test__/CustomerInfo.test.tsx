import React from 'react';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

import { CustomerInfo } from '../CustomerInfo';

const defaultProps = {
  contactPhone: '123456789',
  contactEmail: 'email@email.com'
};

const mountComponent = (customProps: any = {}) => {
  const props = { ...defaultProps, ...customProps };
  const wrapper = shallow(<CustomerInfo {...props} />);
  return { props, wrapper };
};

describe('CustomerInfo component', () => {
  it('Should render header', done => {
    const { wrapper } = mountComponent();
    expect(wrapper).toHaveLength(1);
    done();
  });
  it('should be match with snapshot', done => {
    // Arrange (no Act required)
    const { wrapper } = mountComponent();
    // Assert
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
