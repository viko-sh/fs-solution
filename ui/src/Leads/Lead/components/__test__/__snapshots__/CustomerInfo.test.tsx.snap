// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerInfo component should be match with snapshot 1`] = `ShallowWrapper {}`;
